source/val/validation_state.cpp

```diff
@@ -303,6 +304,12 @@ void ValidationState_t::RegisterCapability(SpvCapability cap) {
   }
 }
 
+void ValidationState_t::RegisterExtension(const char* extension) {
+  if (!strcmp(extension, "SPV_KHR_16bit_storage")) {
+    features_.free_fp_rounding_mode = true;
```

Comment:
The flag should be flipped when we see the corresponding capabilities.
The reason is that a future version of an API can bring the extension into core, and then you'll have modules with the capabilities but not the OpExtension.